Automattic/wp-calypso
# Login: Fix wrong tracking of page views and add missing events

This pull request:

* Adds missing Tracks events in all the login pages
* Renames existing Tracks events in order to be more consistent, and to use the same prefix
* Records error codes in addition to error messages in Tracks events
* Fixes page views tracking
* Fixes a page title that was set [in the wrong place](https://github.com/Automattic/wp-calypso/issues/14430)
* Extracts login footer links into a new component

### Testing Instructions

1. Run `git checkout update/login-tracking` and start your server, or open a [live branch](https://calypso.live/?branch=update/login-tracking)
2. Open the [`Login` page](http://calypso.localhost:3000/log-in/fr?redirect_to=/me) in an incognito window
3. Execute `localStorage.setItem( 'debug', 'calypso:analytics:*' )` in your browser's console
4. Reload the page and assert that a page view is triggered with `/log-in` as path
5. Submit credentials of an 2FA account that has push notifications enabled
6. Assert that a page view is triggered with `/log-in/push` as path
7. Click the `Code via text message` link
8. Assert that a page view is triggered with `/log-in/sms` as path
9. Click the `I can't access my phone` link in the form footer
10. Assert that a page view is triggered with `/log-in/backup` as path
11. Navigate back to the `Login` page
12. Click the `Continue with Google` button
13. Assert that a `calypso_social_login_form_click` event is triggered
14. Assert that its property `social_account_type` is set to `google`

You should also check Tracks to see if these events were correctly recorded.

### Reviews
